### ISSUE TITLE
Add an option for COM812 to disable for single-argument functions?

### ISSUE THREAD

**Issue by davidt** [astral-sh/ruff#18258]
### Summary

I really want to be able to use the COM812 rule, but our codebase contains a ton of code like the following:

```python
some_string = gettext(
    'This is a long string which '
    'is marked for internationalization'
)

another_function(
    some_long +
    expression +
    gets_passed
)
```

While trailing commas do make a lot of sense for function calls that contain multiple arguments, for functions that take a single argument, requiring them muddies the syntax. I'd love to have an option for the flake8-commas plugin (something like `[lint.flake8-commas] allow-single-arg-function-calls = true`)

I'm happy to do the implementation for this, but I wanted to check in first to see if this is something that would be accepted.

**Comment by MichaReiser** [comment]
This seems the same as https://github.com/astral-sh/ruff/issues/13410 and IMO, the same reasoning applies in the sense that this is the exact intention of the rule.

**Comment by davidt** [comment]
Fundamentally, it does look like the same request.

I agree that attempting to do a full static analysis and determine if the callee's signature supports only a single argument is not a good approach, but I'd argue that for the sake of code style, I don't care if there are possible optional arguments (or indeed the potential for possible optional arguments in the future) on the definition side, I only care what's happening at the point of the call.

Adding another argument to such a call would mean needing to add the (now missing) comma, which I agree does affect the diff, but that happens rarely, if ever (gettext's API hasn't changed and I would bet money it never will). In contrast, for things like lists, dicts, or imports, those sorts of changes can happen frequently and optimizing for the diff by requiring the comma makes sense.

Extracting these sorts of args into variables works in some cases, but explicitly does not work with the `gettext()` example, because the scanner that extracts strings into .po files is simple and requires the string to be directly inside the call.

**Comment by davidt** [comment]
I have a working implementation of a setting for this. Should I spend the time to clean up the change and put it up for review?

Thanks!

**Comment by rmlibre** [comment]
I'd just like to chime in here to also express my desire for this option to be made available.

There's another justification for this option which I've not seen expressed before across the many issues and PRs regarding COM812. Which is that parens with a single argument can mean a single-element tuple or just a scope for an expression, depending on whether a comma is or is not present, respectively.

```python
_(
    "this is an implicitly joined string "
    "being passed into a function."
)

_(
    "this is an implicitly joined string "
    "being passed into a function.",
)

(
    "this is an implicitly joined string "
    "which needs the parens syntactically "
    "to create the scope in order for this "
    "to be valid."
)

(
    "this is an implicitly joined string "
    "being set into a single-element "
    "tuple.",
)
```

The first two examples have very little visual distinction, but that's fine because they mean the exact same thing. This is not the case in the later two examples, where the presence of the comma completely changes the meaning of the expression. This subtlety seems to make the lint fix more error-prone for users than even the terminal warning suggests: 

```bash
warning: The following rule may cause conflicts when used with the formatter: `COM812`...
```

because it isn't conflicts with the formatter at issue here, it's the unexpected semantic change. I very much appreciate COM812 for the visual alignment & more localized diffs it provides. But I don't appreciate the added cognitive load of needing to double check that my linter hasn't changed the type of my data while I'm editing.

I would definitely utilize this option if it were made available.
